spring-expression/src/main/java/org/springframework/expression/spel/ast/Indexer.java

```diff
@@ -187,11 +190,72 @@ else if (target instanceof Collection) {
 			return new PropertyIndexingValueRef(
 					target, (String) index, state.getEvaluationContext(), targetDescriptor);
 		}
-
+		Optional<ValueRef> optional = tryIndexAccessor(state, index);
+		if (optional.isPresent()) {
+			return optional.get();
+		}
 		throw new SpelEvaluationException(
 				getStartPosition(), SpelMessage.INDEXING_NOT_SUPPORTED_FOR_TYPE, targetDescriptor);
 	}
 
+	private Optional<ValueRef> tryIndexAccessor(ExpressionState state, Object index) {
+		EvaluationContext context = state.getEvaluationContext();
+		Object target = state.getActiveContextObject().getValue();
+		if (context != null) {
+			List<IndexAccessor> list = context.getIndexAccessors();
+			if (list != null) {
+				List<IndexAccessor> availableAccessors = getIndexAccessorsToTry(state.getActiveContextObject(), list);
+				try {
+					for (IndexAccessor indexAccessor : availableAccessors) {
+						if (indexAccessor.canRead(context, target, index)) {
+							ValueRef valueRef = indexAccessor.read(context, target, index);
+							return Optional.of(valueRef);
+						}
+					}
+				}
+				catch (Exception e) {
+				}
+			}
+		}
+		return Optional.empty();
+	}
+
+	private List<IndexAccessor> getIndexAccessorsToTry(
+			@Nullable Object contextObject, List<IndexAccessor> propertyAccessors) {
+		Class<?> targetType;
```

Comment:
I know [spring code style](https://github.com/spring-projects/spring-framework/wiki/Code-Style#blank-lines) requires one blank line after a method signature that is multiline

Reply:
thanks a lot. I have fixed this problem.

Reply:
Sbrannen's comments show that he probably asked for force push to correct the initial commit. So you don't have to force push every time you add a correction. If you force push all modifications, it is rather difficult for a reviewer to determine what additional modifications are made.